From bfddc49f8119d6718a8a11b130381096ae7c0362 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 24 Apr 2018 18:37:28 -0500
Subject: [PATCH] CMake: Force Multi-threaded DLL runtime

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 660f3b75..84b796fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -215,9 +215,13 @@ IF (MSVC)
   SET(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /w14706")
   # /Oi option enables built-in functions.
   SET(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /Oi")
+  # /MD option for Multi-threaded DLL (non-debug since you cannot mix them).
+  SET(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /MD")
   #################################################################
   # Set compile flags for release build.
   SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /Oi")
+  # /MD option for Multi-threaded DLL.
+  SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /MD")
 ENDIF (MSVC)
 
 # Enable CTest/CDash support
-- 
2.39.3 (Apple Git-146)

